geosnap/analyze/network.py

```diff
@@ -112,24 +154,49 @@ def isochrones_from_id(origin, network, threshold):
     return alpha
 
 
-def isochrones_from_gdf(origins, threshold, network, network_crs=4326, reindex=True):
+def isochrones_from_gdf(
+    origins,
+    threshold,
+    network,
+    network_crs=4326,
+    reindex=True,
+    algorithm="alpha",
+    ratio=0.2,
+    allow_holes=False,
+):
     """Create travel isochrones for several origins simultaneously
 
     Parameters
     ----------
     origins : geopandas.GeoDataFrame
         a geodataframe containing the locations of origin point features
-    network : pandana.Network
-        pandana Network instance for calculating the shortest path isochrone for each origin feature
     threshold: float
-        maximum travel distance to define the isochrone, measured in the same units as edges_df
-        in the pandana.Network object. If the network was created with pandana this is usually meters;
-        if it was created with urbanaccess this is usually travel time in minutes.
-    matrix: pandas dataframe (optional)
-        precalculated adjacency list dataframe created with `compute_travel_adjlist`
+        maximum travel distance to define the isochrone, measured in the same
+        units as edges_df in the pandana.Network object. If the network was
+        created with pandana this is usually meters; if it was created with
+        urbanaccess this is usually travel time in minutes.
+    network : pandana.Network
+        pandana Network instance for calculating the shortest path isochrone
+        for each origin feature
     network_crs : str, int, pyproj.CRS (optional)
-        the coordinate system used to store x and y coordinates in the passed pandana network.
-        If the network was created with pandana or urbanaccess this is nearly always 4326.
+        the coordinate system used to store x and y coordinates in the passed
+        pandana network. If the network was created with pandana or urbanaccess
+        this is nearly always 4326.
+    reindex : bool
+        if True, use the dataframe index as the origin and destination IDs
+        (rather than the node_ids of the pandana.Network). Default is True
+    algorithm : str, {'alpha', 'hull'}
```

Comment:
If the [comment above](https://github.com/oturns/geosnap/pull/398/files#r1510118281) is adopted, we'll also need to update here.